core/api/rpc/wsc.cpp

```diff
@@ -87,10 +88,11 @@ namespace fc::api::rpc {
       }
     }
     chans.clear();
+    reconnect(3, std::chrono::seconds(5));
```

Comment:
Why do we clean everything, and only then try to fix the situation?

Reply:
Also, should we recconect on each error?

Reply:
We clean, because we can't recover session (eg waiting calls and channels), so each waiting call and channel is returned and error, then api caller (eg sealing fsm, or other components) to call again or restore notifications channel.

Reconnecting on connection error allows `ChainNotify` caller to restore notification channel. Also `reconnect()` has mutex